client/starwhale/core/runtime/model.py

```diff
@@ -466,13 +467,50 @@ def venv_install(self, src_dir: Path, env_dir: Path, configs: t.Dict) -> None:
         venv_install_req(env_dir, fpath, pip_config=configs.get("pip"))  # type:ignore
 
 
+class CommandDependency(ASDictMixin, BaseDependency):
+    def __init__(self, deps: t.List[str]) -> None:
+        if isinstance(deps, list):
+            for d in deps:
+                if not isinstance(d, str):
+                    raise FormatError(f"command must be str: {d}")
+        else:
+            raise NoSupportError(
+                f"command dependency only supports list[str] format: {deps}"
+            )
+        self.deps = deps
+
+    @property
+    def kind(self) -> DependencyType:
+        return DependencyType.COMMAND
+
+    def _run_command(self, src_dir: Path, commands: t.List[str]) -> None:
+        for cmd in commands:
+            if (
+                in_container()
+                or os.environ.get("SW_RUNTIME_FORCE_RUN_COMMAND", "0") == "1"
+            ):
+                console.info(f"run command: {cmd}, cwd: {src_dir}")
+                check_call(cmd, shell=True, cwd=src_dir)
```

Comment:
should we add timeout for this command call, or a warning tip? 
`apt-get install xxx` will hang forever